vllm_detector_adapter/generative_detectors/base.py

```diff
@@ -157,9 +158,32 @@ async def process_chat_completion_with_scores(
         logger.debug("Request to chat completion: %s", chat_completion_request)
 
         # Call chat completion
-        chat_response = await self.create_chat_completion(
-            chat_completion_request, raw_request
-        )
+        try:
+            chat_response = await self.create_chat_completion(
+                chat_completion_request, raw_request
+            )
+        except UndefinedError:
+            # Undefined template errors can happen due to a variety of reasons -
+            # e.g. for Granite Guardian it is not limited but including the following
+            # for a particular risk definition: unexpected number of messages, unexpected
+            # ordering of messages, unexpected roles used for particular messages.
+            # Users _may_ be able to correct some of these errors by changing the input
+            # but the error message may not be directly user-comprehensible
+            # This has to be caught first because jinja UndefinedError are TemplateError
+            chat_response = ErrorResponse(
+                message="Template error. Please check request.",
```

Comment:
Does the message in this case not usable to be passed to the user?

Reply:
hmm. true. but that information might still be helpful for debugging. So can we log the raw error message while we switch it with better error message here ?